students/uss_tomasz/lesson_05_lists/5_5/numbers_filter.py

```diff
@@ -0,0 +1,22 @@
+# Problem - Numbers filter
+def strings_list_to_integers_list(strings_list_from_user, filter_range):
+    """Using list comprehensions write a function that casts list of strings to list of integers and
+    filters numbers within supplied range. Example data:
+    list_of_strings = ['2', '0', '8', '3']
+    to_filter_range = range(3)
+    expected_output = [8, 3]
+    """
+    integers_list_for_user = []
+    for element in strings_list_from_user:
+        integers_list_for_user.append(int(element))
+    for i in filter_range:
+        while i in integers_list_for_user:      # The same item can exist multiple times
+            integers_list_for_user.remove(i)
+    return integers_list_for_user
```

Comment:
Although this solution works, you didn't use list comprehension and its filtering capabilities here. Please refactor.